tools/kagami/src/genesis.rs

```diff
@@ -163,15 +164,18 @@ pub fn generate_default(validator_path: Option<PathBuf>) -> color_eyre::Result<R
         .add_parameter(WASM_MAX_MEMORY, DEFAULT_MAX_MEMORY)?
         .into_create_parameters();
 
-    genesis.transactions[0].isi.push(mint.into());
-    genesis.transactions[0].isi.push(mint_cabbage.into());
-    genesis.transactions[0]
-        .isi
-        .push(grant_permission_to_set_parameters.into());
-    genesis.transactions[0].isi.push(parameter_defaults.into());
-    genesis.transactions[0]
-        .isi
-        .push(register_user_metadata_access);
+    let first_tx = genesis
+        .first_transaction_mut()
+        .expect("At least one transaction is expected");
+    for isi in [
+        mint.into(),
+        mint_cabbage.into(),
+        grant_permission_to_set_parameters.into(),
+        parameter_defaults.into(),
+        register_user_metadata_access,
+    ] {
+        first_tx.push_instruction(isi);
+    }
```

Comment:
When how do we say to which transaction we append instruction? Genesis contains transactions, not instructions directly

Reply:
as far as I am aware it always contains only single transaction

Reply:
and in case there are multiple why exactly would we change specifically first transaction of all transactions?

Reply:
> as far as I am aware it always contains only single transaction

Yes, we just always use corner case.

> and in case there are multiple why exactly would we change specifically first transaction of all transactions?

I think there is no real reason.

But having `append_instruction()` in `genesis` does not tell me that it will append instruction to the latest transaction.

Reply:
ok, it's not related to this PR anyhow